src/pages/iou/IOUModal.js

```diff
@@ -142,17 +148,20 @@ class IOUModal extends Component {
             Navigation.dismissModal();
         }
 
-        if (prevProps.myPersonalDetails.preferredCurrencyCode
-            !== this.props.myPersonalDetails.preferredCurrencyCode) {
+        if (prevProps.myPersonalDetails.selectedCurrencyCode
+            !== this.props.myPersonalDetails.selectedCurrencyCode) {
             this.updateSelectedCurrency({
-                currencyCode: this.props.myPersonalDetails.preferredCurrencyCode,
-                currencySymbol: this.props.myPersonalDetails.preferredCurrencySymbol,
+                currencyCode: this.props.myPersonalDetails.selectedCurrencyCode,
+                currencySymbol: this.props.myPersonalDetails.selectedCurrencySymbol,
             });
         }
     }
 
-    getReady() {
-        PersonalDetails.fetchCurrencyPreferences();
+    componentWillUnmount() {
+        Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, {
+            selectedCurrencyCode: this.props.myPersonalDetails.preferredCurrencyCode,
+            selectedCurrencySymbol: this.props.myPersonalDetails.preferredCurrencySymbol,
+        });
```

Comment:
Do other things need to know which currency is selected? Can we just put the "selected" currency on the component level state? Then we would not need to do all of this Onyx stuff.

Reply:
[IOUCurrencySelection](https://github.com/Expensify/Expensify.cash/blob/main/src/pages/iou/IOUCurrencySelection.js#L152-L158) is what actually sets the `selectedCurrency` that is used in `IOUModal`, so we are passing state between components. I'm not sure I see a simple solution that gets rid of Onyx here, but I welcome any ideas.

Reply:
Can we [lift the state up](https://reactjs.org/docs/lifting-state-up.html)?

Reply:
@marcaaron and I discussed 1:1 and lifting state in this case would be tricky because `IOUModal` and `IOUCurrencySelection` share only `ModalStackNavigators` as a common ancestor. We should continue to use Onyx in this case.